TaskLayer/SearchTask/PostSearchAnalysisTask.cs

```diff
@@ -887,5 +887,19 @@ private void WritePeakQuantificationResultsToTsv(FlashLFQResults flashLFQResults
 
             FinishedWritingFile(peaksPath, nestedIds);
         }
+
+        private List<PeptideSpectralMatch> filterPSMList(PostSearchAnalysisParameters parameters, List<PeptideSpectralMatch> psms)
+        {
+            List<PeptideSpectralMatch> list = psms.Where(p => p.FdrInfo.QValue <= CommonParameters.QValueOutputFilter && p.FdrInfo.QValueNotch <= CommonParameters.QValueOutputFilter).ToList();
+            if (!parameters.SearchParameters.WriteDecoys)
+            {
+                list = list.Where(b => !b.IsDecoy).ToList();
```

Comment:
`ToList()` creates a new list in memory, so this is not super RAM efficient to call this 3x here; the first call (line 893) is fine but 896 and 900, try doing `list.RemoveAll(b => b.IsDecoy)` and `list.RemoveAll(b => b.IsContaminant)`